### PR TITLE
uwc: Improvements for sparkplug

### PR DESCRIPTION
**Pull request by aditisa1**
1. Improved the sparkplug code in Main.cpp
2. changed EII to EMB in the codebase of sp and kpi
3. Added concept for time stamp for sp

Signed-off-by: aditisa1 <aditi.saxena@intel.com>

### CODE REVIEW

File: modbus-master/Modbus-App/src/ModbusOnDemandHandler.cpp [hunk 1]
```diff
@@ -1238,13 +1238,13 @@ bool onDemandHandler::processMsg(msg_envelope_t *msg,
 	stMbusApiPram.m_stOnDemandReqData.m_sUsec = getMsgElement(msg, "usec");
 	stMbusApiPram.m_stOnDemandReqData.m_strMqttTime = getMsgElement(msg, "tsMsgRcvdFromMQTT");
 	stMbusApiPram.m_stOnDemandReqData.m_strEiiTime = getMsgElement(msg, "tsMsgPublishOnEII");
-
+	//In case when sparkplug is connected with EMB. 
 	if( stMbusApiPram.m_stOnDemandReqData.m_strMqttTime.empty()){
 		stMbusApiPram.m_stOnDemandReqData.m_strMqttTime = getMsgElement(msg, "tsMsgRcvdFromExtMQTT");
```
Comment (nagdeepgk) [inline modbus-master/Modbus-App/src/ModbusOnDemandHandler.cpp:1243]: Can this be renamed as "tsMsgRcvdFromExtMQTT" TO "tsMsgRcvdFromExtMQTTToSP" ?
Is this naming more appropriate ?

File: sparkplug-bridge/SPARKPLUG-BRIDGE/include/Common.hpp [hunk 2]
```diff
@@ -44,6 +43,8 @@ class CCommon
 private:
 	/** vector of env variables*/
 	std::vector<std::string> m_vecEnv{"AppName", "INTERNAL_MQTT_URL", "DEV_MODE", "NETWORK_TYPE", "DEVICES_GROUP_LIST_FILE_NAME"};
+	// a variable to store the timestamp values of msgs coming from SCADA
```
Comment (nagdeepgk) [inline sparkplug-bridge/SPARKPLUG-BRIDGE/include/Common.hpp:46]: SCADA -> Lets not us3e SCADA term here Let sbe specific if we mean external MQTT/Ignition or what do we mean here.

Comment (nagdeepgk) [inline sparkplug-bridge/SPARKPLUG-BRIDGE/include/Common.hpp:46]: SCADA -> Lets not us3e SCADA term here Let sbe specific if we mean external MQTT/Ignition or what do we mean here.

File: sparkplug-bridge/SPARKPLUG-BRIDGE/include/Common.hpp [hunk 3]
```diff
@@ -44,6 +43,8 @@ class CCommon
...
+	std::string TS_SCADAtoMQTT;
```
Comment (nagdeepgk) [inline sparkplug-bridge/SPARKPLUG-BRIDGE/include/Common.hpp:47]: Again the naming convention needs to be specific "MQTT" meaning is it external etc.. And by SCADA you mean "Sparkplug-bridge" ?

Comment (nagdeepgk) [inline sparkplug-bridge/SPARKPLUG-BRIDGE/include/Common.hpp:47]: Again the naming convention needs to be specific "MQTT" meaning is it external etc.. And by SCADA you mean "Sparkplug-bridge" ?

File: sparkplug-bridge/SPARKPLUG-BRIDGE/include/InternalMQTTSubscriber.hpp [hunk 4]
```diff
@@ -107,8 +107,7 @@ class CIntMqttHandler : public CMQTTBaseHandler
 						metricMapIf_t& a_mapChangedMetrics);
 	bool prepareWriteMsg(std::reference_wrapper<CSparkPlugDev>& a_refSparkPlugDev,
 							metricMapIf_t& a_mapChangedMetrics);
-        bool publish_msg_to_eii(string strPubMsg,string strMsgTopic);
-	bool publish_RD_to_eii(string strPubMsg,string strMsgTopic);
+        bool publish_msg_to_emb(string strPubMsg,string strMsgTopic);
```
Comment (nagdeepgk) [inline sparkplug-bridge/SPARKPLUG-BRIDGE/include/InternalMQTTSubscriber.hpp:110]: Hopefully the CPP has doxygen comments ?

Comment (nagdeepgk) [inline sparkplug-bridge/SPARKPLUG-BRIDGE/include/InternalMQTTSubscriber.hpp:110]: Hopefully the CPP has doxygen comments ?

Comment (aditisa1) [reply sparkplug-bridge/SPARKPLUG-BRIDGE/include/InternalMQTTSubscriber.hpp:110]: yes , it is added in CPP

**Comment by aditisa1** [comment]
Closing this one because of git issue , opened following one 
https://github.com/open-edge-insights/uwc/pull/56/files

**Pull request closed without merging**